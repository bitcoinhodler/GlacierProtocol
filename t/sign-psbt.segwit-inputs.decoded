{
  "txid": "0fe2bd16b8edc16cd19ba7bcc298dcc367fec094b4a365fed7a70daf35d60f3a",
  "hash": "0fe2bd16b8edc16cd19ba7bcc298dcc367fec094b4a365fed7a70daf35d60f3a",
  "version": 2,
  "size": 1358,
  "vsize": 1358,
  "weight": 5432,
  "locktime": 0,
  "vin": [
    {
      "txid": "185b60ce1e42b86658f746e3c2b17d71ee60e87c4fca2ea073bf8ae2b117a080",
      "vout": 1,
      "scriptSig": {
        "asm": "0 304402205803f381fb75cc407f69d4a0d09daafd63c58b8c55a1ac544007489a8a017c5402200db11a426c921f6ed96a4c3c6841db5567365c28b093c533cdb281c829fa19b1[ALL] 30440220371f52039f2c941e87ca5e1c16f51aed72f4c94e647e8d088d48ccc48a556c1a02203c06d4ec3f1ba137a3dc39a1bd4cb82230e39d6391c74f6f20c164830977feb7[ALL] 522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae",
        "hex": "0047304402205803f381fb75cc407f69d4a0d09daafd63c58b8c55a1ac544007489a8a017c5402200db11a426c921f6ed96a4c3c6841db5567365c28b093c533cdb281c829fa19b1014730440220371f52039f2c941e87ca5e1c16f51aed72f4c94e647e8d088d48ccc48a556c1a02203c06d4ec3f1ba137a3dc39a1bd4cb82230e39d6391c74f6f20c164830977feb7014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      },
      "sequence": 4294967295
    },
    {
      "txid": "f3ac0558c10f2559eb1a95a9e94c557596255c6e65e71b230a7d3fcb8a8da06c",
      "vout": 0,
      "scriptSig": {
        "asm": "0 30440220485ff92d1f25e0f3c7ad9723f066ea69615d07c86b6f27e3deadcbd625ee4563022022265c500fbc53fbd9bcf7b210776bb2d81b0ccb8556d23b6219fb61f303915d[ALL] 304402206ac3fdcf9de9c21b2f6e459f386b892f52a7662ad85022f94e9d2078c511002302204f525a5719904929db699643ac072b20f37b226d3245f6e6d2a6feecb98037ca[ALL] 522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae",
        "hex": "004730440220485ff92d1f25e0f3c7ad9723f066ea69615d07c86b6f27e3deadcbd625ee4563022022265c500fbc53fbd9bcf7b210776bb2d81b0ccb8556d23b6219fb61f303915d0147304402206ac3fdcf9de9c21b2f6e459f386b892f52a7662ad85022f94e9d2078c511002302204f525a5719904929db699643ac072b20f37b226d3245f6e6d2a6feecb98037ca014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      },
      "sequence": 4294967295
    },
    {
      "txid": "96ffed5227118919b9a13a0d977010196acfb934660529e1cf5e51829de42a21",
      "vout": 0,
      "scriptSig": {
        "asm": "0 304402200e4414d70049025a0f7b3b057b177dd25a5738f7420f32509731ae48fd0db41f02205e4d8cac5723d9e2b9709d79ddc401f8d57e21f1d5d688e8ed915e089ab7ae53[ALL] 304402205374af637d8cd42a989aab6b97c0b82600f3d25ded5f17758f15556fb5626696022000ce4dda76813929470ecd2278ebb1113db8833a6457fb835fea51d65634920a[ALL] 522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae",
        "hex": "0047304402200e4414d70049025a0f7b3b057b177dd25a5738f7420f32509731ae48fd0db41f02205e4d8cac5723d9e2b9709d79ddc401f8d57e21f1d5d688e8ed915e089ab7ae530147304402205374af637d8cd42a989aab6b97c0b82600f3d25ded5f17758f15556fb5626696022000ce4dda76813929470ecd2278ebb1113db8833a6457fb835fea51d65634920a014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      },
      "sequence": 4294967295
    },
    {
      "txid": "5c0b2602310ac978105fc6c9c1f2d3d9d6bbaa69669400f6a9ae37d3ea406025",
      "vout": 0,
      "scriptSig": {
        "asm": "0 3044022044d70809c9d4f2e5545700af1f2689d7956d039786dc4d31271661a1c93203340220017202f916274a5bf330d3d7f6031ffe29afa4e2e0618d0e39e5a3ed6d2a10b5[ALL] 304402200cff1e3c9c69b2114ad3970b310521b7b17857bff6de277f6c19c3777000a958022005b746d86df4674c922f222709bf6da5561df9d22238d6ee81d6469a4b7e6aa6[ALL] 522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae",
        "hex": "00473044022044d70809c9d4f2e5545700af1f2689d7956d039786dc4d31271661a1c93203340220017202f916274a5bf330d3d7f6031ffe29afa4e2e0618d0e39e5a3ed6d2a10b50147304402200cff1e3c9c69b2114ad3970b310521b7b17857bff6de277f6c19c3777000a958022005b746d86df4674c922f222709bf6da5561df9d22238d6ee81d6469a4b7e6aa6014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 1.69997793,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 8546e747c8094376b0a0af44bc5bbb8dd4f0218d OP_EQUAL",
        "desc": "addr(2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow)#tjjjpqgh",
        "hex": "a9148546e747c8094376b0a0af44bc5bbb8dd4f0218d87",
        "address": "2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow",
        "type": "scripthash"
      }
    }
  ]
}
