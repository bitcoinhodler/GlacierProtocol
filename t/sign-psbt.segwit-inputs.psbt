cHNidP8BAM4CAAAABKHSDZTq99J5Uap9bAnOKi32p0tjpAjTzHSM1faFQJh2AQAAAAD/////Pu5kUHA2R7Bllp0XVS+jx25RPu0kblMwuBGQRAHLw6gAAAAAAP////846dka8Cz8TInqLX3J0nqp1WVNxrKQW9ndWmlRRCcqNQAAAAAA/////+TSpo+sFp17+/OHKGoGDXiWDSjtH33vGL8Wns52ulr3AAAAAAD/////AeH1IQoAAAAAF6kUhUbnR8gJQ3awoK9EvFu7jdTwIY2HAAAAAAABAIoCAAAAAcYRLayDnIjY87BCwy/4aqU0C7aLtMGlmuH2z9U3pvSbAQAAABcWABT52vlSVEpC2f45XTnl/dJHLpWFE/////8CEH6IAgAAAAAXqRTm/PfGtbPE2+MPUGJDz9+Ko20ygYegf4gCAAAAABepFL05lruabEAdp+ZdTbs6m5eM0EXshwAAAAABBItSIQPRTdz7aBf1V5aVu7PrPhhYh78pQrAx5vcWNDuP5+no4iECj81G+GFLLL8xgJaWgkKh4ivPttjyttyTnIwnw0eyk3shAxWstVASD0zctGDVxJCAq1CMpMzY3t7Kwi/urIzQF9TBIQIrBj7i8i+eGYLBQP53hnLGgxEaeW3Mzb1Hu2Xj1gipg1SuAAEAcgIAAAABfq8Wi5V9w6CzBakhyTy6/xkfZ6NYBdIKWvGz6UAlgPoBAAAAAP////8CCEuHAgAAAAAXqRS9OZa7mmxAHafmXU27OpuXjNBF7Idws4kCAAAAABYAFNovQSpVal/ZGPcVN377sXqJEN2CAAAAAAEEi1IhA9FN3PtoF/VXlpW7s+s+GFiHvylCsDHm9xY0O4/n6ejiIQKPzUb4YUssvzGAlpaCQqHiK8+22PK23JOcjCfDR7KTeyEDFay1UBIPTNy0YNXEkICrUIykzNje3srCL+6sjNAX1MEhAisGPuLyL54ZgsFA/neGcsaDERp5bczNvUe7ZePWCKmDVK4AAQBqAgAAAAE8/CDWAf1nqjaqRPJPI4iuJOu1ru9gkRxXMTNROdd0LgEAAAAXFgAUUdSJGWfS4DgB1hnaf87L/UO7gaj/////AYh9iAIAAAAAF6kUvTmWu5psQB2n5l1Nuzqbl4zQReyHAAAAAAEEi1IhA9FN3PtoF/VXlpW7s+s+GFiHvylCsDHm9xY0O4/n6ejiIQKPzUb4YUssvzGAlpaCQqHiK8+22PK23JOcjCfDR7KTeyEDFay1UBIPTNy0YNXEkICrUIykzNje3srCL+6sjNAX1MEhAisGPuLyL54ZgsFA/neGcsaDERp5bczNvUe7ZePWCKmDVK4AAQBTAgAAAAERZUM7BcTw6EKIf4Ra65K+XGi2ADozpP27H7akGLjehgEAAAAA/////wH/sokCAAAAABepFL05lruabEAdp+ZdTbs6m5eM0EXshwAAAAABBItSIQPRTdz7aBf1V5aVu7PrPhhYh78pQrAx5vcWNDuP5+no4iECj81G+GFLLL8xgJaWgkKh4ivPttjyttyTnIwnw0eyk3shAxWstVASD0zctGDVxJCAq1CMpMzY3t7Kwi/urIzQF9TBIQIrBj7i8i+eGYLBQP53hnLGgxEaeW3Mzb1Hu2Xj1gipg1SuAAA=
