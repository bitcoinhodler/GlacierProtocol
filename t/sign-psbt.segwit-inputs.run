#!/bin/bash
set -e

# Test withdrawing from transactions that were created by SegWit wallets.
# I have four transactions to try:
# 1. Electrum p2wpkh-in-p2sh with change
# 2. Electrum native p2wpkh with change
# 3. Electrum p2wpkh-in-p2sh with no change
# 4. Electrum native p2wpkh with no change


$GLACIERSCRIPT --regtest=$1 sign-psbt << INPUT 2>&1
y
y
y
y
y
y
../../t/sign-psbt.segwit-inputs.psbt
y
2
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4
y
INPUT
