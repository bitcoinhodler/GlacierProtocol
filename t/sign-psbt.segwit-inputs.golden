Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?Input a filename located in the current directory which contains the PSBT:
1.69999151 BTC in unspent inputs from cold storage address 2NAVkgszAvjZZwWpHpbCaxSaNqCCXPD8Bts
1.69997793 BTC going to destination address 2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow
Fee amount: 0.00001358
Confirm? (y/n): 
How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: 
Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

1.69999151 BTC in unspent inputs from cold storage address 2NAVkgszAvjZZwWpHpbCaxSaNqCCXPD8Bts
1.69997793 BTC going to destination address 2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow
Fee amount: 0.00001358

Signing with private keys: 
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 1.00 satoshis per vbyte

Raw signed transaction (hex):
0200000004a1d20d94eaf7d27951aa7d6c09ce2a2df6a74b63a408d3cc748cd5f68540987601000000fd1e010047304402203636f0fade964147f9f73d44f418859e5d0080d4cd9d78d5f27ed6a885348e700220266a7c25b6d6b704b8fcb99629d78d011d9b8507d1c01f1ef37dfec5c6c0fdbc014730440220604999f47ba05c243dda0f34246e5eb1fd58b98116f3ab72e5e139b24bb4d528022072828656c98f50d095161383eafd86daac01c0025c769acaddaf6198333260d2014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354aeffffffff3eee6450703647b065969d17552fa3c76e513eed246e5330b811904401cbc3a800000000fd1e010047304402203825edb4a15afc0faabca49f60a62cc60ce1e3b0e71ca6bdf1e14d1f873db495022024f42c858324fe792bf8d4535576b41136c6ce605398f5c3e213025df4f6d1c80147304402206e2e7f540a03b2799f05ef318ac6d83057be90b49407707c52fdbc1a0cc00a2702206b2837485d890fbdea7423c7cf60900183bb2bed03ce00a18edbdcd719980842014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354aeffffffff38e9d91af02cfc4c89ea2d7dc9d27aa9d5654dc6b2905bd9dd5a695144272a3500000000fd1e010047304402205ce4e8c3463b371ba5f96c2679405730c6cb855c22987258c72d704be6a2a26502205f72ccc6018a1229dbe40088e233307d158ee71dd6217bbbfe7af6ab72037333014730440220735d9024077000b819f2baaec5a25e8f1f51c38e782e38f07faa7ecf20e37d1f02204bcb42b465dc6ca0681cf26f2f144565fabaf308b66d1e64f79d161306644ea8014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354aeffffffffe4d2a68fac169d7bfbf387286a060d78960d28ed1f7def18bf169ece76ba5af700000000fd1e0100473044022077efa6ec04e34de55523c4c7fd49362dd610be21a0db678354c31d29f80103cb022017060a344bfaef3db06953aad4eb2a4761ea7179837183e5314f9d24745043a201473044022067fbbd1b82927561f1c75bc1cb7fdfa6b9307d6e774baca81fab36c73e201dbd022048ec801eaa906e28ccb37906a98bf0275a39f0a1033815cb8cab26503f6a170b014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354aeffffffff01e1f5210a0000000017a9148546e747c8094376b0a0af44bc5bbb8dd4f0218d8700000000

Transaction fingerprint (md5):
b113c61b950c8b99cfbc080ddc00ef4e
QR code for transaction written to transaction.png
